src/index.ts

```diff
@@ -444,16 +453,25 @@ export class Supertab {
         amount: tab.limit,
         text: priceToText(tab.limit),
       },
-      currency: tab.currency,
+      currency: getPublicCurrencyDetails(currencyObject),
       paymentModel: tab.paymentModel,
       purchases: tab.purchases.map((purchase) => {
+        if (purchase.price.currency !== currencyObject.isoCode) {
+          throw new Error(
+            "Currency mismatch: Expected tab currency " +
+              tab.currency +
+              " and purchase currency " +
+              purchase.price.currency +
+              " to be equal.",
+          );
+        }
```

Comment:
Isn't this unnecessary? It seems like we're checking for something that's never going to happen. 🤔 The same tab currency is enforced on backend at the time of making a purchase. There's no way to have EUR tab including USD purchases as the `/v1/purchase` will respond with error when you attempt to add an item of different currency than tab currency and the purchase won't be added to a tab.

Reply:
Hi Tom, good point! I put the check in there because the code now makes an assumption about how Tapper handles currencies in tabs and tab purchases. I think the performance and maintenance cost of this check is relatively small. On the other hand it might save us some headaches in the future if/when Tapper behavior changes (who knows?).

Reply:
Alright, I'm fine with that.